indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -239,4 +241,57 @@ UNIT_TEST(EditableMapObject_GetNamesDataSource)
                 "The English language + user`s language. Excluding repetiton"));
   }
 }
+
+UNIT_TEST(EditableMapObject_SetInternet)
+{
+  try
+  {
+    classificator::Load();
+  }
+  catch (RootException const & e)
+  {
+    LOG(LERROR, ("Classificator read error: ", e.what()));
+  }
+
+  EditableMapObject emo;
+  const uint32_t wifiType = classif().GetTypeByPath({"internet_access", "wlan"});
+  emo.SetType(wifiType);
+
+  auto types = emo.GetTypes();
+  TEST(types.Has(wifiType), ());
+
+  auto const setInternetAndCheck = [wifiType](EditableMapObject & emo, osm::Internet internet, bool wifiExpected)
+  {
+    emo.SetInternet(internet);
+
+    TEST_EQUAL(emo.GetInternet(), internet, ());
+    auto const & types = emo.GetTypes();
+    TEST_EQUAL(types.Has(wifiType), wifiExpected, ());
+  };
+
+  setInternetAndCheck(emo, osm::Internet::No, false);
+  setInternetAndCheck(emo, osm::Internet::Yes, false);
+  setInternetAndCheck(emo, osm::Internet::Wired, false);
+  setInternetAndCheck(emo, osm::Internet::Wlan, true);
+  setInternetAndCheck(emo, osm::Internet::Unknown, false);
+  setInternetAndCheck(emo, osm::Internet::No, false);
+  setInternetAndCheck(emo, osm::Internet::Wlan, true);
+
+
+  EditableMapObject bunkerEmo;
+  bunkerEmo.SetType(classif().GetTypeByPath({"military", "bunker"}));
+  types = bunkerEmo.GetTypes();
+  TEST(!types.Has(wifiType), ());
+
+  setInternetAndCheck(bunkerEmo, osm::Internet::Wlan, true);
```

Comment:
What the purpose of this group of checks? What's the difference between these checks and checks for emo?